pkg/pdatatest/plogtest/options.go

```diff
@@ -0,0 +1,180 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plogtest // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/plogtest"
+
+import (
+	"bytes"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
+)
+
+// LogsCompareOption can be used to mutate expected and/or actual logs before comparing.
+type LogsCompareOption interface {
```

Comment:
Same, the Logs prefix seem a bit redundant.

Reply:
Currently, this option is applicable to `CompareLogs` only (supposed to be called `CompareLogsOption` btw), `CompareScopeLogs` and `CompareLogRecords` go with no options at this point. If we want to add options to those functions later, we will need `CompareScopeLogsOption` and `CompareLogRecordsOption`. That's why I think it's better to be explicit with the naming like `<function_name>Option`. Let me know WDYT

Reply:
> should to be called CompareLogsOption btw

100% agree, prefix with the func name.